storm-client/src/jvm/org/apache/storm/metric/api/IDimensionalMetric.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The ASF licenses this file to you under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package org.apache.storm.metric.api;
+
+import java.util.Map;
+
+public interface IDimensionalMetric extends IMetric {
+
+    /**
+     * Dimensions is a collection of additional key-value metadata which is optional.
+     * It could be useful for further fine-grained measurement for a metric.
+     * It may also help better grouping in downstream metric monitoring system.
+     *
+     * @return a map of dimension name to a sub-metric
+     */
+    Map<Map<String, String>, CountMetric> getDimToMetric();   // dimensions : sub-metric
```

Comment:
Using `CountMetric` makes this interface specific to `CountMetric`.

Reply:
@kishorvpatil My mistake, thank you for pointing out. My intention was to set this to IMetric. When implementing, map value could be dynamically typed by implementing IMetric. Corrected.